src/pages/TeamDashboard/teamdashboard.js

```diff
@@ -12,20 +14,27 @@ query getTeam($id: Int!) {
 `
 
 const TeamDashboard = (props) => {
-  const id = Number(props.match.params.id)
+  const id = Number(props.match.params.id);
+  const { isLoggedIn } = useSelector(state => state.auth);
   const { loading, error, data } = useQuery(GET_TEAM_INFO, {
     variables: { id }
   })
   console.log("This is the team data: ", data)
   if (loading) return 'Loading...'
   if (error) return `Error! ${error.message}`
-  return (
-    <Fragment>
-      <h1>This is the Team Dashboard page</h1>
-      <h2>The Team Using this has ID {id}</h2>
-      <h2>This is their data: {JSON.stringify(data.getTeam)}</h2>
-    </Fragment>
-  )
+  if (isLoggedIn){
+    return (
+      <Fragment>
+        <h1>This is the Team Dashboard page</h1>
+        <h2>The Team Using this has ID {id}</h2>
+        <h2>This is their data: {JSON.stringify(data.getTeam)}</h2>
+      </Fragment>
+    )
+  } else {
+    return(
+      <Redirect to="/login"/>
+    );
```

Comment:
Is there a way to use PrivateRoute in App.js to handle these routes instead of adding a redirect for all components?